docs/observability.md

```diff
@@ -0,0 +1,69 @@
+# OpenTelemetry Service Observability
+
+## Goal
+
+The goal of this document is to have a comprehensive description of observability of the Service and changes needed to achieve observability part of our [vision](vision.md).
+
+## What Needs Observation
+
+The following elements of the Service need to be observable.
+
+### Current Values
+
+- Resource consumption: CPU, RAM (in the future also IO - if we implement persistent queues) and any other metrics that may be available to Go apps (e.g. garbage size, etc).
+- Receiving data rate, broken down by receivers and by data type (traces/metrics).
+- Exporting data rate, broken down by exporters and by data type (traces/metrics).
+- Data drop rate due to throttling, broken down by data type.
+- Data drop rate due to invalid data received, broken down by data type.
+- Current throttling state: Not Throttled/Throttled by Downstream/Internally Saturated.
+- Incoming connection count, broken down by receiver.
+- Incoming connection rate (new connections per second), broken down by receiver.
+- In-memory queue size (in bytes and in units).
+- Persistent queue size (when supported).
+- End-to-end latency (from receiver input to exporter output). Note that with multiple receivers/exporters we potentially have NxM data paths, each with different latency (plus different pipelines in the future), so realistically we should likely expose the average of all data paths (perhaps broken down by pipeline).
+- Latency broken down by pipeline elements (including exporter network roundtrip latency for request/response protocols).
+
+“Rate” values must reflect the average rate of the last 10 seconds. Rates must exposed in bytes/sec and units/sec (e.g. spans/sec).
```

Comment:
I think we can defer the calculation of rates to metric backend. The service observability can simply record and export cumulative values.

Reply:
Yes, that's also what is suggested in the text as a possible option / open question:

>Note: some of the current values may be calculated as derivatives of cumulative values in the backend, so it is an open question if we want to expose them separately or no.

Reply:
One of the reasons when rates may be desirable to be calculated in the service is when you have a deployment where services are not monitored using a metrics monitoring system. Own calculations may not be as good as a full-blown metrics backend can provide but it may be better than nothing to have the rates dumped to service's log file periodically.

Reply:
@songy23 let me know if you think the text is unclear and you want changes there.

Reply:
The current statement looks good :)